### DISCUSSION TITLE
blog/2024/11/04/render-generated-r-chunks-quarto/

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#100] [Blog comments]
# blog/2024/11/04/render-generated-r-chunks-quarto/

Andrew Heiss is an assistant professor at Georgia State University, researching international NGOs and teaching data science & economics.

https://www.andrewheiss.com/blog/2024/11/04/render-generated-r-chunks-quarto/

**Comment by ericmshearer** [comment]
Great write-up, thanks! I've started testing out targets for my own work. I do wonder how you were able to make the percentage bars in the results table (second image from the top, results table, "District 16").

**Reply by andrewheiss** [reply]
Thanks! Those tables were made with [{reactablefmtr}](https://kcuilla.github.io/reactablefmtr/index.html):

```
library(dplyr)
library(reactable)
library(reactablefmtr)

data_for_table <- mtcars |> 
  group_by(cyl) |> 
  summarize(avg_mpg = mean(mpg))

data_for_table |> 
  reactable(
    columns = list(
        avg_mpg = colDef(
        cell = data_bars(
          data_for_table, 
          text_position = "above",
          round_edges = T,
          number_fmt = scales::label_comma(accuracy = 0.1)
        )
      )
    )
  )
```

<img width="378" alt="image" src="https://github.com/user-attachments/assets/e40aefeb-f6dd-4888-97f1-f3e449be160b">

**Comment by DagHjermann** [comment]
Thanks a lot for yet another super useful blog post! And the election website is really nice too.

**Comment by jbengler** [comment]
Hi Andrew,

thank you for the great post!
I have one suggestion to remove duplication ;)

```
p1_text <- glue("
  ggplot(gapminder_2007, aes(x = gdpPercap, y = lifeExp, color = continent)) +
    geom_point()")
p2_text <- glue("
  {p1_text} +
    scale_x_log10(labels = scales::dollar_format(accuracy = 1))
")
p3_text <- glue('
  {p2_text} +
    scale_color_viridis_d(option = "plasma", end = 0.9)
')
p4_text <- glue("
  {p3_text} +
    theme_minimal()
")
p5_text <- glue('
  {p4_text} +
    labs(x = "GDP per capita", y = "Life expectancy", color = "Continent")
')

plot_text <- list(p1_text, p2_text, p3_text, p4_text, p5_text)

plot_list <- tribble(
  ~code_text, ~description,
  p1_text, "Start with the initial plot…",
  p2_text, "…use a logarithmic x-axis…",
  p3_text, "…change the color palette…",
  p4_text, "…change the theme…",
  p5_text, "…and change the default labels"
) %>% 
  rowwise() %>% 
  mutate(plot = list(eval(parse(text = code_text))))
```
